sources/packages/web/src/views/institution/locations/programs/LocationPrograms.vue

```diff
@@ -47,45 +68,114 @@
+import { InstitutionService } from "@/services/InstitutionService";
+import { InstitutionRoutesConst } from "@/constants/routes/RouteConstants";
+import {
+  ClientIdType,
+  EducationProgramsSummaryPaginated,
+  DEFAULT_PAGE_LIMIT,
+  DEFAULT_PAGE_NUMBER,
+  PAGINATION_LIST,
+  InstitutionProgramSummaryFields,
+  DataTableSortOrder,
+} from "@/types";
 import { ref, watch, onMounted } from "vue";
+import StatusBadge from "@/components/generic/StatusBadge.vue";
+import { useFormatStatuses } from "@/composables";
 
 export default {
+  components: { StatusBadge },
   props: {
     locationId: {
       type: Number,
       required: true,
     },
-    locationName: {
-      type: String,
-      required: true,
-    },
   },
   setup(props: any) {
     const router = useRouter();
-    const programs = ref([] as SummaryEducationProgramDto[]);
+    const programAndCount = ref({} as EducationProgramsSummaryPaginated);
+    const { getProgramStatusToGeneralStatus } = useFormatStatuses();
+    const locationDetails = ref();
+    const loading = ref(false);
+    const searchBox = ref("");
+    const currentPage = ref();
+    const currentPageLimit = ref();
 
-    const loadSummary = async () => {
-      programs.value = await EducationProgramService.shared.getLocationProgramsSummary(
+    /**
+     * function to load program list and count for institution
+     * @param page page number, if nothing passed then DEFAULT_PAGE_NUMBER
+     * @param pageCount page limit, if nothing passed then DEFAULT_PAGE_LIMIT
+     * @param sortField sort field, if nothing passed then InstitutionProgramSummaryFields.ApprovalStatus
+     * @param sortOrder sort oder, if nothing passed then DataTableSortOrder.ASC
+     */
+    const loadSummary = async (
+      page = DEFAULT_PAGE_NUMBER,
+      pageCount = DEFAULT_PAGE_LIMIT,
+      sortField = InstitutionProgramSummaryFields.ApprovalStatus,
+      sortOrder = DataTableSortOrder.ASC,
+    ) => {
+      loading.value = true;
+      programAndCount.value = await EducationProgramService.shared.getLocationProgramsSummary(
         props.locationId,
+        page,
+        pageCount,
+        searchBox.value,
+        sortField,
+        sortOrder,
       );
+      loading.value = false;
     };
 
-    onMounted(loadSummary);
-    watch(() => props.locationId, loadSummary);
+    // pagination sort event callback
+    const paginationAndSortEvent = async (event: any) => {
+      currentPage.value = event?.page;
+      currentPageLimit.value = event?.rows;
+      await loadSummary(
+        event.page,
+        event.rows,
+        event.sortField,
+        event.sortOrder,
+      );
+    };
+    const loadProgramDetails = async () => {
+      locationDetails.value = await InstitutionService.shared.getInstitutionLocation(
+        props.locationId,
+      );
+    };
+    // search program table
+    const searchProgramTable = async () => {
+      await loadSummary(
+        currentPage.value ?? DEFAULT_PAGE_NUMBER,
```

Comment:
we can use || instead for the advantage of compatible browsers

Reply:
In my understanding this is a concern while doing form.io but not for Vue or any modern UI framework. Please let me know if we need further explanations here.

Reply:
@andrewsignori-aot once we had a discussion on NOA story, we noticed that Nullish coalescing operator is not supported in IE and Opera. While || is supported for all browsers.

![image](https://user-images.githubusercontent.com/54600590/152399731-f45b5621-1995-49ab-b3cc-c89c4bd21e59.png)